server/api/jurisdictions.py

```diff
@@ -535,6 +537,64 @@ def status(jurisdiction_id: str) -> JurisdictionStatus:
     }
 
 
+@api.route(
+    "/election/<election_id>/jurisdictions/last-login",
+    methods=["GET"],
+)
+@restrict_access([UserType.AUDIT_ADMIN])
+def get_last_login_by_jurisdiction(election: Election):
+    # Get all jurisdiction IDs for this election
+    jurisdiction_ids = [
+        jurisdiction_id
+        for (jurisdiction_id,) in Jurisdiction.query.with_entities(Jurisdiction.id)
+        .filter_by(election_id=election.id)
+        .all()
+    ]
+
+    # Look for events after most recent round started, or election creation time
+    # if no rounds exist yet
+    query_timestamp_after = election.created_at
+    round = (
+        Round.query.filter_by(election_id=election.id)
+        .order_by(Round.created_at.desc())
+        .one_or_none()
+    )
+    if round is not None:
+        query_timestamp_after = round.created_at
+
+    # Query for login activities from users who are related to jurisdictions we found earlier
+    activities = (
+        ActivityLogRecord.query.join(
+            User, ActivityLogRecord.info["base"]["user_key"].astext == User.email
+        )
+        .join(User.jurisdictions)
+        .filter(
+            Jurisdiction.id.in_(jurisdiction_ids),
```

Comment:
It might be a bit more efficient to just filter by the jurisdiction's `election_id` field

```py
.join(User.jurisdictions)
.filter(Jurisdiction.election_id == election.id)
```

Reply:
https://github.com/votingworks/arlo/pull/2061/commits/1552349d670efb58c45844de06fda8cea005718d#diff-fc2514bfe998fc28ba94dbd7860ae951626cf2ddb9a6aeb3bbd29f7e67cd4cdaR559